src/main/scala/is/hail/methods/Aggregators.scala

```diff
@@ -371,6 +371,50 @@ class SumArrayAggregator[T](implicit ev: scala.math.Numeric[T], ct: ClassTag[T])
   def copy() = new SumArrayAggregator()
 }
 
+class MaxAggregator[T, BoxedT >: Null](implicit ev: NumericPair[T, BoxedT], ct: ClassTag[T]) extends TypedAggregator[BoxedT] {
+
+  import ev.numeric
+  import Ordering.Implicits._
+
+  var _state: BoxedT = _
```

Comment:
why can't _state be unboxed if you've got both the primitive and boxed types hanging around?